code/datums/spells/rathens.dm

```diff
@@ -0,0 +1,57 @@
+/obj/effect/proc_holder/spell/aoe_turf/rathens
+	name = "Rathen's Secret"
+	desc = "Summons a powerful shockwave around you that tears the appendix and limbs off of enemies."
+	charge_max = 500
+	clothes_req = 1
+	invocation = "ARSE NATH!"
+	invocation_type = "shout"
+	range = 7
+	cooldown_min = 200
+	selection_type = "view"
+	action_icon_state = "superfart"
+
+/obj/effect/proc_holder/spell/aoe_turf/rathens/cast(list/targets)
+	var/mob/user = usr
+	var/list/impacted_mobs = list()
+	for(var/turf/T in targets)
```

Comment:
You could make this spell be a subtype of `/obj/effect/proc_holder/spell/targeted` with `max_targets = 0` and then `targets` would be the list of mobs `impacted_mobs` you're manually generating here. It would eliminate lines 14-24.